### PR TITLE
apps: add unit test cases to cover more device remove cases

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Cover the cases where a device being removed has bricks,
and the case where the device has bricks but nowhere to put them.

Signed-off-by: John Mulligan <jmulligan@redhat.com>

### CODE REVIEW

**Comment by obnoxxx** [comment]
@phlogistonjohn rebase needed

**Comment by phlogistonjohn** [comment]
Rebased

**Comment by phlogistonjohn** [comment]
Fixed go fmt issues (I keep forgetting to run the check in my excitement to push things), and rebased.

File: apps/glusterfs/device_entry_test.go [hunk 1]
```diff
@@ -719,3 +722,216 @@ func TestDeviceSetStateOfflineOnline(t *testing.T) {
...
+		3,    // devices_per_node,
+		5*TB, // disksize
+	)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	vreq := &api.VolumeCreateRequest{}
+	vreq.Size = 100
+	vreq.Durability.Type = api.DurabilityReplicate
+	vreq.Durability.Replicate.Replica = 3
+	// create two volumes
```
Comment (obnoxxx) [inline apps/glusterfs/device_entry_test.go]: I'm counting five...

File: apps/glusterfs/device_entry_test.go [hunk 2]
```diff
@@ -719,3 +722,216 @@ func TestDeviceSetStateOfflineOnline(t *testing.T) {
...
+	for i := 0; i < 5; i++ {
+		v := NewVolumeEntryFromRequest(vreq)
+		err = v.Create(app.db, app.executor, app.Allocator())
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	}
+
+	// grab a devices that has bricks
```
Comment (obnoxxx) [inline apps/glusterfs/device_entry_test.go]: s/devices/device/

File: apps/glusterfs/device_entry_test.go [hunk 3]
```diff
@@ -719,3 +722,216 @@ func TestDeviceSetStateOfflineOnline(t *testing.T) {
...
+}
+
+func TestDeviceSetStateFailedTooFewDevices(t *testing.T) {
+	tmpfile := tests.Tempfile()
+	defer os.Remove(tmpfile)
+
+	app := NewTestApp(tmpfile)
+	defer app.Close()
+	//	router := mux.NewRouter()
+	//	app.SetRoutes(router)
```
Comment (obnoxxx) [inline apps/glusterfs/device_entry_test.go]: what about those 2 commented-out lines?

File: apps/glusterfs/device_entry_test.go [hunk 4]
```diff
@@ -719,3 +722,216 @@ func TestDeviceSetStateOfflineOnline(t *testing.T) {
...
+		1,    // devices_per_node,
+		5*TB, // disksize
+	)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	vreq := &api.VolumeCreateRequest{}
+	vreq.Size = 100
+	vreq.Durability.Type = api.DurabilityReplicate
+	vreq.Durability.Replicate.Replica = 3
+	// create two volumes
```
Comment (obnoxxx) [inline apps/glusterfs/device_entry_test.go]: two?

File: apps/glusterfs/device_entry_test.go [hunk 5]
```diff
@@ -719,3 +722,216 @@ func TestDeviceSetStateOfflineOnline(t *testing.T) {
...
+	for i := 0; i < 5; i++ {
+		v := NewVolumeEntryFromRequest(vreq)
+		err = v.Create(app.db, app.executor, app.Allocator())
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	}
+
+	// grab a devices that has bricks
```
Comment (obnoxxx) [inline apps/glusterfs/device_entry_test.go]: devices->device

File: apps/glusterfs/device_entry_test.go [hunk 6]
```diff
@@ -719,3 +722,216 @@ func TestDeviceSetStateOfflineOnline(t *testing.T) {
...
+				return err
+			}
+			if len(d.Bricks) > 0 {
+				return nil
+			}
+		}
+		t.Fatalf("should have at least one device with bricks")
+		return nil
+	})
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
```
Comment (obnoxxx) [inline apps/glusterfs/device_entry_test.go]: we might want a function for this. but can be done later...

**Comment by phlogistonjohn** [comment]
Updated wrt review comments.

**Pull request merged**